examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -0,0 +1,308 @@
+
+        # not sure if I want to print these...
+        # print("before: ", before)
+        # print("after: ", after)
+
+    # TODO: also save dl_dp for the other parameter types we're not necessarily updating, for later analysis
+    #   (would be easier with syntax like forcefield[handle_type])
+
+    return parameter_updates
+
+
+def _save_forcefield(filename, ff_params):
+    # TODO: update path
+
+    with open(path_to_results.joinpath(filename), 'w') as fh:
+        fh.write(ff_params)
+
+
+if __name__ == "__main__":
+    # which force field components we'll refit
+    forces_to_refit = [nonbonded.AM1CCCHandler, nonbonded.LennardJonesHandler]
+
+    # how much computation to spend per refitting step
+    # configuration = testing_configuration  # a little
+    # configuration = production_configuration  # a lot
+    configuration = intermediate_configuration  # goldilocks
+
+    # how many parameter update steps
+    num_parameter_updates = 1000
+    # TODO: make this configurable also
+
+    # set up multi-GPU client
+    client = CUDAPoolClient(max_workers=configuration.num_gpus)
+
+    # load pre-defined collection of relative transformations
+    with open(path_to_transformations, 'rb') as f:
+        relative_transformations: List[RelativeFreeEnergy] = load(f)
+
+    # update the forcefield parameters for a few steps, each step informed by a single free energy calculation
+
+    # compute and save the sequence of relative_transformation indices
+    num_epochs = int(np.ceil(num_parameter_updates / len(relative_transformations)))
+    np.random.seed(2021)
+    step_inds = []
+    for epoch in range(num_epochs):
+        inds = np.arange(len(relative_transformations))
+        np.random.shuffle(inds)
+        step_inds.append(inds)
+    step_inds = np.hstack(step_inds)[:num_parameter_updates]
+    np.save(path_to_results.joinpath('step_indices.npy'), step_inds)
+
+    # build the complex system
+    complex_system, complex_coords, _, _, complex_box, _ = builders.build_protein_system(
+        path_to_protein)
+    # TODO: optimize box
+    complex_box += np.eye(3) * 0.1  # BFGS this later
+
+    # build the water system
+    solvent_system, solvent_coords, solvent_box, _ = builders.build_water_system(4.0)
+    # TODO: optimize box
+    solvent_box += np.eye(3) * 0.1  # BFGS this later
+
+    # note: "complex" means "protein + solvent"
+    binding_model = RBFEModel(
+        client=client,
+        ff=forcefield,
+        complex_system=complex_system,
+        complex_coords=complex_coords,
+        complex_box=complex_box,
+        complex_schedule=construct_lambda_schedule(configuration.num_complex_windows),
+        solvent_system=solvent_system,
+        solvent_coords=solvent_coords,
+        solvent_box=solvent_box,
+        solvent_schedule=construct_lambda_schedule(configuration.num_solvent_windows),
+        equil_steps=configuration.num_equil_steps,
+        prod_steps=configuration.num_prod_steps,
+    )
+
+    import jax
```

Comment:
Did you want to import this here?

Reply:
Resolved in https://github.com/proteneer/timemachine/pull/331/commits/9dacdd8384c6b4678a2449715eaf8ba3a47a1fa0